fix path to git clone when using ssh

shahzebsiddiqui opened the pull request:
Address bug #293 

```
(buildtest) siddiq90@DOE-7086392 buildtest % buildtest get git@github.com:buildtesters/tutorials.git
Cloning into '/Users/siddiq90/.buildtest/site/github.com/buildtesters/tutorials'...
remote: Enumerating objects: 106, done.
remote: Counting objects: 100% (106/106), done.
remote: Compressing objects: 100% (73/73), done.
remote: Total 106 (delta 32), reused 97 (delta 25), pack-reused 0
Receiving objects: 100% (106/106), 20.97 KiB | 3.00 MiB/s, done.
Resolving deltas: 100% (32/32), done.
```

shahzebsiddiqui committed:
fix path to git clone when using ssh

vsoch reviewed (changes requested):
Now the normal url doesn’t work? Please use a regex that can detect either, and parse pieces accordingly. Many of our users don’t have a key configured so both options are needed.

shahzebsiddiqui committed:
increase coverage for buildtest get command.
Add tests for checking func_get_subcmd for validating input parameters,
testing http link, ssh link, and running some edge cases where error is raised.

shahzebsiddiqui committed:
skip ssh test when run in Travis

vsoch commented on buildtest/menu/get.py:
```diff
@@ -33,9 +33,14 @@ def func_get_subcmd(args):
+    if re.search("^(git@github.com)", args.repo):
+        url = args.repo.split(":")[1]
+        username = url.split("/")[0]
+
```
Should there be any checks to see if something already exists at the clone path? Should the clone happen if this is the case?

shahzebsiddiqui replied:
There is coverage when trying to reclone an existing repo that resolves to same path. See coverage report for get.py at https://coveralls.io/builds/30956049/source?filename=buildtest/menu/get.py

The check in framework looks fine, if dest directory exists we raise SystemError.

shahzebsiddiqui replied:
The reclone test is found at https://github.com/shahzebsiddiqui/buildtest-1/blob/0b9bf6f1448df6c0def7fee848b38a84ba25d74a/tests/menu/test_get.py#L30
